### PR TITLE
[nginx] define resolver directive in the config

### PR DESCRIPTION
**Pull request by mikz**
so cosockets use internal resolver and don't have to
go through APIcast resolver.

Closes https://github.com/3scale/apicast/issues/488

### CODE REVIEW

File: gateway/src/apicast/cli/environment.lua [hunk 1]
```diff
@@ -14,10 +14,26 @@ local assert = assert
 local error = error
 local print = print
 local pairs = pairs
+local ipairs = ipairs
+local tostring = tostring
 local insert = table.insert
 local concat = table.concat
 local re = require('ngx.re')
 
+local function parse_nameservers()
```
Comment (davidor) [inline gateway/src/apicast/cli/environment.lua]: Shouldn't this method be exposed in resty.resolver?

Comment (mikz) [reply gateway/src/apicast/cli/environment.lua]: resty.resolver exposes the nameservers as tables with some metamethods. Those objects can be joined with table.join so they have to be serialized first into `ip:port`. I guess it is worth a comment in the code. Will fix it.

Comment (mikz) [reply gateway/src/apicast/cli/environment.lua]: Fixed.

File: gateway/src/apicast/cli/environment.lua [hunk 2]
```diff
@@ -14,10 +14,32 @@ local assert = assert
...
+    local resolver = require('resty.resolver')
+    local nameservers = {}
+
+    for _,nameserver in ipairs(resolver.init_nameservers()) do
+        -- resty.resolver returns nameservers as tables with __tostring metamethod
+        -- unfrotunately those objects can't be joined with table.concat
```
Comment (davidor) [inline gateway/src/apicast/cli/environment.lua]: typo :eyes:

Comment (mikz) [reply gateway/src/apicast/cli/environment.lua]: ⭕️ Ⓜ️ :godmode:

File: gateway/src/apicast/cli/environment.lua (hunk 1 above)
Comment (davidor) [reply gateway/src/apicast/cli/environment.lua]: That helps :+1:

**Pull request merged**